examples/all-clusters-app/all-clusters-common/include/WhmDelegate.h

```diff
@@ -0,0 +1,270 @@
+     *
+     *********************************************************************************/
+
+    /**
+     * @brief Set the Water Header Mode and act accordingly.
+     *
+     * @param mode  The Water Heater Mode (e.g. OFF, MANUAL or TIMED).
+     */
+    Protocols::InteractionModel::Status SetWaterHeaterMode(uint8_t mode);
+
+    /**
+     * @brief Set the water temperature of the tank
+     *
+     * @param waterTemperature  The water temperature in 100th's Celsius
+     */
+    void SetWaterTemperature(uint16_t waterTemperature);
+
+    /**
+     * @brief Set the target water temperature of the tank
+     *
+     * @param targetWaterTemperature  The water temperature in 100th's Celsius
+     */
+    void SetTargetWaterTemperature(uint16_t targetWaterTemperature);
+
+    /**
+     * @brief Determine whether the heating sources need to be turned on or off
+     */
+    Protocols::InteractionModel::Status CheckIfHeatNeedsToBeTurnedOnOrOff();
+
+    /**
+     * @brief Static timer callback for when Boost timer expires.
+     */
+    static void BoostTimerExpiry(System::Layer * systemLayer, void * delegate);
+
+    /**
+     * @brief Object timer callback for when Boost timer expires.
+     */
+    void HandleBoostTimerExpiry();
+
+    /**
+     * Determines whether the tank water temperature has reached the target temperature.
+     *
+     * @return Returns True is tank water temperature has reached the target temperature, False otherwise.
+     */
+    bool HasWaterTemperatureReachedTarget() const;
+
+    /**
+     * Simulates water being drawn from the water tank.
+     *
+     * @param percentageReplaced  The % of water being replaced with water with a temperature of replacedWaterTemperature.
+     * @param replacedWaterTemperature  The temperature of the percentageReplaced water.
+     */
+    void DrawOffHotWater(uint8_t percentageReplaced, uint16_t replacedWaterTemperature);
+
+private:
+    /**
+     * @brief Determine whether heating needs to be turned on or off or left as is.
+     *
+     * @param heatingOp[out]  Set as determined whether heating needs to be turned on/off or left unchanged.
+     *
+     * @return  Success if the heating operation could be determined otherwise returns with appropriate error.
+     */
+    Protocols::InteractionModel::Status DetermineIfChangingHeatingState(HeatingOp & heatingOp);
+
+private:
+    /*********************************************************************************
+     *
+     * WaterHeaterManagementDelegate specific attributes
+     *
+     *********************************************************************************/
+
+    // Need the following so can determine which features are supported
+    WaterHeaterManagement::Instance * mpWhmInstance;
+
+    // Pointer to the manufacturer specific object which understand the hardware
+    WhmManufacturer * mpWhmManufacturer;
+
+    // Target water temperature in 100ths of a C
+    uint16_t mTargetWaterTemperature;
```

Comment:
I understand this is needed for the logic of the implementation, but given that the thermostat cluster would be present in a real application, shouldn't this be in the implementation rather than making it so that each implementation inherits those temperature members?

Reply:
Added this to https://github.com/project-chip/connectedhomeip/issues/34565 follow up issue.